pkg/apis/workflow/v1alpha1/workflow_phase.go

```diff
@@ -0,0 +1,22 @@
+package v1alpha1
+
+// the workflow's phase
+type WorkflowPhase string
+
+const (
+	WorkflowUnknown   WorkflowPhase = ""
+	WorkflowPending   WorkflowPhase = "Pending" // pending some set-up - rarely used
+	WorkflowRunning   WorkflowPhase = "Running" // any node has started; pods might not be running yet, the workflow maybe suspended too
+	WorkflowSucceeded WorkflowPhase = "Succeeded"
+	WorkflowFailed    WorkflowPhase = "Failed" // it maybe that the the workflow was terminated
+	WorkflowError     WorkflowPhase = "Error"
+)
+
+func (p WorkflowPhase) Completed() bool {
```

Comment:
Should this be fulfilled?

Reply:
I'm not sure either fulfilled or complete should apply to the workflows phase at all. The source of truth is the label. Phase and conditions are just copies of this information.